onyo/lib/onyo.py

```diff
@@ -1066,9 +1100,13 @@ def set(self, paths: Iterable[Union[Path, str]], values: dict, dryrun: bool,
         if name_values:
             try:
                 self._update_names(assets_to_set, name_values)
-                self.clear_caches()
             except ValueError as e:
-                self.clear_caches()
+                self.clear_caches(assets=True,  # `onyo set` can rename assets
+                                  dirs=False,  # no szenario with `set`
+                                  files=True,  # in theory might change other YAML
+                                  gitfiles=False,  # no szenario with `set`
+                                  templates=True  # might modify templates
+                                  )
                 if self.files_staged:
                     self.restore()
                 # reset renaming needs double-restoring
```

Comment:
Why?!?

Reply:
Ping

Reply:
Because, when using the command we use to restore (`git restore --source=HEAD --staged --worktree`) it does not clean up the tree completely, it just reverses the renaming, and then it does need another `git restore` to get the (through the moving) deleted file back.
```
❱ git status
On branch master
nothing to commit, working tree clean
❱ git mv repair/laptop_apple_macbookpro.dd082o recycling/new_name_same.device
❱ git status
On branch master
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	renamed:    repair/laptop_apple_macbookpro.dd082o -> recycling/new_name_same.device

tkadelka@bnbmac73 in ~/onyo on git:master+
❱ git restore --source=HEAD --staged --worktree recycling/new_name_same.device
tkadelka@bnbmac73 in ~/onyo on git:master+
❱ git status
On branch master
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	deleted:    repair/laptop_apple_macbookpro.dd082o
```

I did back in the days when working on this not find any other solution to reverse it completely with one call that also works for all other commands. If you know or come along an alternative please let me know!